optuna/_gp/optim_mixed.py

```diff
@@ -0,0 +1,309 @@
+from __future__ import annotations
+
+import math
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+from optuna._gp.acqf import AcquisitionFunctionParams
+from optuna._gp.acqf import eval_acqf_no_grad
+from optuna._gp.acqf import eval_acqf_with_grad
+from optuna._gp.search_space import normalize_one_param
+from optuna._gp.search_space import sample_normalized_params
+from optuna._gp.search_space import ScaleType
+from optuna.logging import get_logger
+
+
+if TYPE_CHECKING:
+    import scipy.optimize as so
+else:
+    from optuna import _LazyImport
+
+    so = _LazyImport("scipy.optimize")
+
+_logger = get_logger(__name__)
+
+
+def _local_search_continuous(
+    acqf_params: AcquisitionFunctionParams,
+    initial_params: np.ndarray,
+    initial_fval: float,
+    continuous_params: np.ndarray,
+    continuous_param_scale: np.ndarray,
+    tol: float,
+) -> tuple[np.ndarray, float] | None:
+    if len(continuous_params) == 0:
+        return None
+    normalized_params = initial_params.copy()
+
+    def negfun_continuous_with_grad(x: np.ndarray) -> tuple[float, np.ndarray]:
+        normalized_params[continuous_params] = x * continuous_param_scale
+        (fval, grad) = eval_acqf_with_grad(acqf_params, normalized_params)
+        # Flip sign because scipy minimizes functions.
+        return (-fval, -grad[continuous_params] * continuous_param_scale)
+
+    x_opt, fval_opt, info = so.fmin_l_bfgs_b(
+        func=negfun_continuous_with_grad,
+        x0=normalized_params[continuous_params] / continuous_param_scale,
+        bounds=[(0, 1 / s) for s in continuous_param_scale],
+        pgtol=math.sqrt(tol),
+        maxiter=200,
+    )
+
+    if info["warnflag"] == 2:
+        fval_opt = negfun_continuous_with_grad(x_opt)[0]
```

Comment:
[Question] What is the difference between `fval_opt` above and the `fval_opt` here?

```suggestion
        # Maximum number of iterations reached.
        fval_opt = negfun_continuous_with_grad(x_opt)[0]
```

Reply:
It's just ensuring that `fval_opt` is consistent with `x_opt`, because `l-bfgs-b` may not return a correct `fval_opt` or `x_opt` when an error (info["warnflag"] == 2) happens.

Reply:
It seems that this line is unnecessary. I'll remove it.